pkg/loki/loki.go

```diff
@@ -111,6 +113,10 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&c.UseBufferedLogger, "log.use-buffered", true, "Uses a line-buffered logger to improve performance.")
 	f.BoolVar(&c.UseSyncLogger, "log.use-sync", true, "Forces all lines logged to hold a mutex to serialize writes.")
 
+	//TODO(trevorwhitney): flip this to false with Loki 3.0
+	f.BoolVar(&c.LegacyReadTarget, "legacy-read-mode", true, "Set to false to disable the legacy read mode and use new scalable mode with 3rd backend target. "+
```

Comment:
Is there some validation for `legacy-read-mode=true` and `target=backend`?

Reply:
yes: https://github.com/trevorwhitney/loki/blob/5135a4325d78cd2e2e29f503c1ca9b54ad9a51d5/pkg/loki/loki.go#L644